src/makeCalculator.js

```diff
@@ -4,7 +4,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate: function (operation, number = 0) {
+      operation.call(this, number);
+
+      return this;
+    },
+
+    add(number) {
+      this.result += number;
+
+      return this;
+    },
+
+    subtract(number) {
+      this.result -= number;
+
+      return this;
+    },
+
+    divide(number) {
+      this.result /= number;
+
+      return this;
+    },
+
+    multiply(number) {
+      this.result *= number;
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+  };
 }
```

Comment:
The `makeCalculator` function is well structured and adheres to the naming convention as per the checklist. However, you can improve the `operate` method by using the shorthand method syntax instead of the `function` keyword.